### PR TITLE
Enforce consistent type imports with ESLint

### PR DESCRIPTION
**Pull request by nwalters512**
As discussed here: https://github.com/PrairieLearn/PrairieLearn/pull/10804#discussion_r1801185547

### CODE REVIEW

File: apps/prairielearn/src/tests/sync/util.ts [hunk 1]
```diff
@@ -4,7 +4,7 @@ import { assert } from 'chai';
 import stringify from 'fast-json-stable-stringify';
 import fs from 'fs-extra';
 import * as tmp from 'tmp-promise';
-import { z } from 'zod';
+import { type z } from 'zod';
```
Comment (jonatanschroeder) [inline apps/prairielearn/src/tests/sync/util.ts:7]: This is an interesting one, though it makes sense based on its use in this context.

File: packages/express-test-utils/src/index.ts [hunk 2]
```diff
@@ -1,6 +1,6 @@
-import { Server } from 'node:http';
+import { type Server } from 'node:http';
 
-import * as express from 'express';
+import type * as express from 'express';
```
Comment (jonatanschroeder) [inline packages/express-test-utils/src/index.ts:3]: I wonder if it makes sense to import the type `Express` here instead? Nit.

File: packages/session/src/test-utils.ts [hunk 3]
```diff
@@ -1,6 +1,6 @@
-import { Server } from 'node:http';
+import { type Server } from 'node:http';
 
-import express from 'express';
+import type express from 'express';
```
Comment (jonatanschroeder) [inline packages/session/src/test-utils.ts:3]: Again, maybe import the actual type?

**Review by jonatanschroeder — APPROVED** [review]

File: packages/express-test-utils/src/index.ts (hunk 2 above)
Comment (nwalters512) [reply packages/express-test-utils/src/index.ts]: Done in ad94972.

File: packages/session/src/test-utils.ts (hunk 3 above)
Comment (nwalters512) [reply packages/session/src/test-utils.ts]: Done in ad94972.

**Pull request merged**